src/toggle/toggle.tsx

```diff
@@ -228,29 +229,44 @@ export const Toggle = ({
                 $styleType={styleType}
             >
                 <IndicatorLabelContainer $addPadding={removable}>
-                    <Input
-                        ref={inputRef}
-                        name={name}
-                        id={`${generatedId}-input`}
-                        type={type === "checkbox" ? "checkbox" : "radio"}
-                        data-testid="toggle-input"
-                        disabled={disabled}
-                        onChange={handleOnChange}
-                        checked={selected}
-                        aria-describedby={ariaDescriptions}
-                        {...otherProps}
-                    />
-                    {indicator && renderIndicator()}
-                    <TextContainer $selected={selected} $disabled={disabled}>
-                        <Label
-                            htmlFor={`${generatedId}-input`}
-                            data-testid={`${generatedId}-toggle-label`}
-                            $maxLines={childrenMaxLines}
+                    <InputContainer
+                        id={`${generatedId}-input-container`}
+                        onClick={() => {
+                            inputRef?.current?.click();
+                        }}
+                    >
+                        <Input
+                            ref={inputRef}
+                            name={name}
+                            id={`${generatedId}-input`}
+                            type={type === "checkbox" ? "checkbox" : "radio"}
+                            data-testid="toggle-input"
+                            disabled={disabled}
+                            onChange={handleOnChange}
+                            onClick={(e) => {
+                                e.stopPropagation();
+                            }}
+                            checked={selected}
+                            aria-describedby={ariaDescriptions}
+                            aria-labelledby={`${generatedId}-toggle-label`}
```

Comment:
actually this is not needed since we're still using the `<label>` element with `htmlFor`, so the input and label are still associated